### PR TITLE
Feature/dev 36 header refactoring

### PR DESCRIPTION
**Pull request by RC-3222**
Transformed header into a common composite component (so that it could use parts from (profile-related info from users) and be used in (AuthLayout in auth) other modules ) & added support for auth pages

image

### CODE REVIEW

**Review by AAKlimkov — APPROVED** [review]

File: src/components/header/navigation/main-navigation.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    UserOutlined,
+} from "@ant-design/icons";
+
+import { routes } from "@/router/constants";
+
+import styles from "./navigation.module.scss";
+
+interface NavigationProps {
+    isDrawerOpen: boolean;
+    setDrawerOpen: (arg1: boolean) => void;
```
Comment (shev7) [inline src/components/header/navigation/main-navigation.tsx]:
```suggestion
    setDrawerOpen: Dispatch<SetStateAction<boolean>>;
```

File: src/components/header/navigation/main-navigation.tsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        <Drawer
+            placement="left"
+            closable={false}
+            onClose={onClose}
+            open={isDrawerOpen}
+            width={250}
+            styles={{
+                header: { padding: "10px", backgroundColor: "#2E2E2E" },
+                body: { padding: 0 },
+            }}
```
Comment (shev7) [inline src/components/header/navigation/main-navigation.tsx]: style using classname, aboid using inline styles

**Review by shev7 — APPROVED** [review]

File: src/components/header/navigation/main-navigation.tsx (hunk 1 above)
Comment (RC-3222) [reply src/components/header/navigation/main-navigation.tsx]: fixed

File: src/components/header/navigation/main-navigation.tsx (hunk 2 above)
Comment (RC-3222) [reply src/components/header/navigation/main-navigation.tsx]: fixed

**Review by savelievamareena — APPROVED** [review]

**Pull request merged**